### PR TITLE
Handle errors logically

### PR DESCRIPTION
**Pull request by naggingant**

### CODE REVIEW

File: combiner.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			}
+		}
+
+		for k, v := range part.Data {
+			retResponse.Data[k] = v
+		}
+	}
+
+	if retStatus > http.StatusInternalServerError {
+		retStatus = http.StatusInternalServerError
```
Comment (PuKoren) [inline combiner.go]: `if isComplete == false` maybe also ? There may be one path of code where `resStatus` is still 0 here

Comment (PuKoren) [reply combiner.go]: Ha that might be covered in `jsonErrorRender` :bowing_man:

Comment (naggingant) [reply combiner.go]: @PuKoren I think this is something we should seriously consider though, the krakend responds with status 200 and with a possibly incomplete body when at least one of the backend hosts succeeded, which is different from what we discussed.
Didn't think about this well...

Comment (naggingant) [reply combiner.go]: Will keep this in mind, for when it causes some problem.

**Pull request merged**